code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/catalog/impl/CatalogServiceImpl.java

```diff
@@ -247,13 +249,15 @@ protected void refresh() throws IOException {
         if (firstCall) {
             String url = CATALOG_RESOURCE_URL.get();
             if (url.endsWith("/")) {
-                url = url.substring(0, url.length()-1);
-            }
-            Response resp = Request.Post(String.format("%s?refresh&action=refresh", url)).execute();
-            int status = resp.returnResponse().getStatusLine().getStatusCode();
-            if (status >= 400) {
-                throw new IOException("Failed to reload got [" + status + "]");
+                url = url.substring(0, url.length() - 1);
             }
+            Request.Post(String.format("%s?refresh&action=refresh", url)).execute().handleResponse((resp) -> {
+                int status = resp.getStatusLine().getStatusCode();
+                if (status >= 400) {
+                    log.warn("Skipping catalog refresh because it failed. Response code: {}, reason: {}", status, resp.getStatusLine().getReasonPhrase());
+                }
+                return null;
+            });
             firstCall = false;
```

Comment:
Not sure how firstCall parameter is used, just wanted to make sure that it is being set to false even though the catalog refresh had failed, is acceptable.